frame/pov-limit/src/migrations.rs

```diff
@@ -0,0 +1,59 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Migrations for Pov-Limit Pallet
+
+use super::*;
+use frame_support::{pallet_prelude::StorageVersion, traits::OnRuntimeUpgrade};
+
+pub mod v1 {
+	use super::*;
+
+	pub struct MigrateToV1<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> OnRuntimeUpgrade for MigrateToV1<T> {
```

Comment:
This is an initialization? Aka when you want to put the pallet onto a running chain? We should not have this here as a migration. This should probably be some chain specific code that is written when the pallet is added.

Reply:
That makes sense, but I think there is a problem with this. Before benchmarking `on_idle` we have to initialize `TrashData` so that it is not empty. If we were to make the initialization code chain specific how could we initialize data inside the benchmark? I cannot call the function because I cannot know where it will be located and what the name will be since this would be implemented chain specific.

Reply:
>This should probably be some chain specific code that is written when the pallet is added.  

What would that code do? Currently we only have to  populate a storage map to make the pallet work. That is not very "chain specific".

Reply:
Ok, I will remove the migration then and put the code inside the kitchensink runtime

Reply:
> I think a "compromise" would be to add a function initialize_pallet that does what this "migration" is doing

Do you mean by this to create an extrinsic that is callable by root origin only if the `TrashData` is empty, or?

Reply:
Yes. In general having a `Controller` origin would be better, which can also set the computation limits.

Reply:
What would the `Controller` origin be in this context, would it be a custom origin?